x-pack/plugins/cloud_integrations/cloud_chat/server/plugin.ts

```diff
@@ -27,7 +28,12 @@ export class CloudChatPlugin implements Plugin<void, void, CloudChatSetupDeps> {
   }
 
   public setup(core: CoreSetup, { cloud, security }: CloudChatSetupDeps) {
-    if (cloud.isCloudEnabled && this.config.chatIdentitySecret) {
+    if (
+      cloud.isCloudEnabled &&
+      this.config.chatIdentitySecret &&
+      cloud.trialEndDate &&
+      isTodayInDateWindow(cloud.trialEndDate, this.config.trialBuffer)
+    ) {
```

Comment:
nit: If Kibana starts when the trial (+ buffer) isn't due, the route will be registered AND it will be still registered after that time if Kibana is never restarted. Is that intentional?

Reply:
Hmmm... good point...I should add a `badRequest` to the route.  Nice catch.

I should probably just change this to return a bad request instead.